manim/utils/unit.py

```diff
@@ -18,21 +19,86 @@ def __rmul__(self, val):
 
 
 class Percent:
-    def __init__(self, axis):
+    """A percentage (0-100) scaling class of each axis.
+
+    Parameters
+    ----------
+    axis
+        The axis (X or Y) of the frame to consider the percentage of.
+        Z_axis has no such frame limits so can't be used here.
+
+    Examples
+    --------
+
+    .. manim :: PercentAxisExample
+        :ref_classes: Axes Dot ApplyMethod
+
+        class PercentAxisExample(Scene):
+            def construct(self):
+                axes = Axes(
+                    x_range=[-1, 0, .1],
+                    y_range=[-1, 0, .1],
+                    x_length=100 * unit.Percent(X_AXIS),
+                    y_length=100 * unit.Percent(Y_AXIS),
+                    axis_config={"color": YELLOW},
+                    tips=False
+                )
+                self.add(axes)
+
+                dot = Dot(radius=0.2, color=RED)
+
+                self.play(ApplyMethod(dot.shift, 30 * unit.Percent(X_AXIS) * RIGHT))
+                self.play(ApplyMethod(dot.shift, 30 * unit.Percent(Y_AXIS) * UP))
+                # using 30 and LEFT below achieves the same effect
+                self.play(ApplyMethod(dot.shift, -30 * unit.Percent(X_AXIS) * RIGHT))
+                # using 30 and DOWN below achieves the same effect
+                self.play(ApplyMethod(dot.shift, -30 * unit.Percent(Y_AXIS) * UP))
+
+    """
+
+    def __init__(self, axis: np.ndarray):
         if np.array_equal(axis, constants.X_AXIS):
             self.length = config.frame_width
         if np.array_equal(axis, constants.Y_AXIS):
             self.length = config.frame_height
         if np.array_equal(axis, constants.Z_AXIS):
             raise NotImplementedError("length of Z axis is undefined")
 
-    def __mul__(self, val):
+    def __mul__(self, val: float) -> float:
         return val / 100 * self.length
 
-    def __rmul__(self, val):
+    def __rmul__(self, val: float) -> float:
         return val / 100 * self.length
 
 
 Pixels = _PixelUnits()
+"""A scaling object to convert pixels to Munits
+
+    Examples
+    --------
+
+    .. manim :: PixelsExample
+        :ref_classes: Dot ApplyMethod
+
+        class PixelsExample(Scene):
+            def construct(self):
+                dot1 = Dot(radius=0.2, color=RED).shift(UP)
+                dot2 = Dot(radius=0.2, color=BLUE).shift(DOWN)
+                self.play(
+                    # With the standard Manim frame width/ratio,
+                    # 150 Pixels = 2.5 Munits
```

Comment:
This comment is a bit problematic in so far, as this depends on the selected render quality -- and we use rather particular setting for the documentation that is actually different from all of the other presets (`medium_quality`, `high_quality` etc) that people usually use. Not sure whether there is a good solution that works both in a rendered form in the documentation and with our usual presets. 🤔

Reply:
My thinking here is that perhaps I could just explicitly state the conversion used by the docs with a disclaimer that this changes based on quality settings.
For example, something similar to:
```python
# With the low quality (-ql) Manim frame width/ratio,
#     150 Pixels = 2.5 Munits
# This ratio will change depending on your resolution, quality configuration etc.
```
Where would I find the documentation's Manim configuration?
I tried doing some digging in `\docs\` and `\docs\source\` while the documentation was building locally but haven't been able to find anything yet.

If the configuration is entirely unwieldy as you suggest (with some weird pixel-munit ratio), I might just remove the manim directive entirely and leave it as a plaintext example.

Reply:
Alternatively, I could just include the `quality` option (see commit below)
```python
    .. manim :: PixelsExample
        :quality: low
        :ref_classes: Dot ApplyMethod
```